tracer_test.go

```diff
@@ -143,13 +147,13 @@ func (s *tracerSuite) TestTraceStartedOrJoinedMetrics() {
 		s.Equal(test.sampled, sp1.Context().(SpanContext).IsSampled())
 		s.Equal(test.sampled, sp2.Context().(SpanContext).IsSampled())
 
-		assertMetrics(s.T(), s.stats, []expectedMetric{
-			{[]string{"jaeger.spans", "group", "sampling", "sampled", test.label}, 3},
-			{[]string{"jaeger.spans", "group", "lifecycle", "state", "started"}, 3},
-			{[]string{"jaeger.spans", "group", "lifecycle", "state", "finished"}, 3},
-			{[]string{"jaeger.traces", "state", "started", "sampled", test.label}, 1},
-			{[]string{"jaeger.traces", "state", "joined", "sampled", test.label}, 1},
-		})
+		testutils.AssertCounterMetrics(s.T(), s.metricsFactory, []testutils.ExpectedMetric{
+			{Name: "jaeger.spans", Tags: map[string]string{"group": "sampling", "sampled": test.label}, Value: 3},
+			{Name: "jaeger.spans", Tags: map[string]string{"group": "lifecycle", "state": "started"}, Value: 3},
+			{Name: "jaeger.spans", Tags: map[string]string{"group": "lifecycle", "state": "finished"}, Value: 3},
+			{Name: "jaeger.traces", Tags: map[string]string{"sampled": test.label, "state": "started"}, Value: 1},
+			{Name: "jaeger.traces", Tags: map[string]string{"sampled": test.label, "state": "joined"}, Value: 1},
```

Comment:
this still became a lot more verbose than before, but ok.

One way to solve that would be to have a constructor
```
func NewExpectedMetric(name string, value int, tags ...string)
```
then we wouldn't need to repeat field names and map[string]string, but just use variadic syntax both in the constructor and the overall AssertCounterMetric function

It's a nice to have.